internal/services/dataprotection/data_protection_backup_policy_postgresql_resource.go

```diff
@@ -185,6 +173,146 @@ func resourceDataProtectionBackupPolicyPostgreSQL() *pluginsdk.Resource {
 			},
 		},
 	}
+
+	if !features.FourPointOhBeta() {
+		// remove 'default_retention_duration' in tf 4.0 is replaced by `default_retention_rule.0.life_cycle.#.duration`
+		resource.Schema["default_retention_duration"] = &pluginsdk.Schema{
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ISO8601Duration,
+		}
+		// remove 'duration' in tf 4.0 is replaced by `retention_rule.#.life_cycle.#.duration`
+		resource.Schema["retention_rule"].Elem.(*pluginsdk.Resource).Schema["duration"] = &pluginsdk.Schema{
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ISO8601Duration,
+		}
+	} else {
+		// Add `life_cycle` in tf 4.0 since `retention_rule.#.life_cycle.#.target_copy_setting` and `retention_rule.#.life_cycle.#.data_store_type` should be specified by user and are not fixed.
+		resource.Schema["retention_rule"].Elem.(*pluginsdk.Resource).Schema["life_cycle"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"data_store_type": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ForceNew: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							// confirmed with the service team that the possible values do not include `OperationalStore`.
+							string(backuppolicies.DataStoreTypesVaultStore),
+							string(backuppolicies.DataStoreTypesArchiveStore),
+						}, false),
+					},
+
+					"duration": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validate.ISO8601Duration,
+					},
+
+					"target_copy_setting": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						ForceNew: true,
+						MaxItems: 1,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"copy_after": {
```

Comment:
What does this property do exactly?

Reply:
Specify when the backups are tiered across two or more selected data stores.

Reply:
that means this is a duration? should be call it `copy_after_duration`

Reply:
@katbyte thanks for your suggestion. `copy_after` really does not make it clear what this feature means. I assume that `copy_option` might be more suitable as the meaning of this feature is `when` to back up across two of more selected data stores. For example:  Setting Immediate Copy means having a backup copy in both the standard vault and the archive vault simultaneously. Setting Copy On Expiry means moving the backup to vault-archive upon its expiry in vault-standard. WDYT?